src/globus_sdk/services/auth/client/service_client.py

```diff
@@ -441,28 +444,23 @@ def get_identity_providers(
 
         log.debug("Looking up Globus Auth Identity Providers")
 
-        if query_params is None:
-            query_params = {}
-
-        if domains is not None and ids is not None:
+        if domains is not MISSING and ids is not MISSING:
             raise exc.GlobusSDKUsageError(
                 "AuthClient.get_identity_providers does not take both "
                 "'domains' and 'ids'. These are mutually exclusive."
             )
-        # if either of these params has a truthy value, stringify it
-        # this handles lists of values as well as individual values gracefully
-        # letting us consume args whose `__str__` methods produce "the right
-        # thing"
-        elif domains is not None:
-            query_params["domains"] = commajoin(domains)
-        elif ids is not None:
-            query_params["ids"] = commajoin(ids)
-        else:
+        elif domains is MISSING and ids is MISSING:
             log.warning(
                 "neither 'domains' nor 'ids' provided to get_identity_providers(). "
```

Comment:
This is getting out-of-scope for this PR, but since we're here:

```suggestion
                "Neither 'domains' nor 'ids' provided to get_identity_providers(). "
```

Reply:
I'll allow it! :stuck_out_tongue_winking_eye: